trieste/acquisition/utils.py

```diff
@@ -159,6 +159,29 @@ def copy_to_local_models(
     return {LocalizedTag(key, i): copy.deepcopy(global_model) for i in range(num_local_models)}
 
 
+def add_local_datasets(
```

Comment:
The name suggests that you're updating the input in place (though the type hint says otherwise). Maybe rename?
 
```suggestion
def with_local_datasets(
```